java/vector/src/main/java/org/apache/arrow/vector/types/TimeUnit.java

```diff
@@ -19,10 +19,10 @@
 package org.apache.arrow.vector.types;
 
 public enum TimeUnit {
-  SECOND(org.apache.arrow.flatbuf.TimeUnit.SECOND),
-  MILLISECOND(org.apache.arrow.flatbuf.TimeUnit.MILLISECOND),
-  MICROSECOND(org.apache.arrow.flatbuf.TimeUnit.MICROSECOND),
-  NANOSECOND(org.apache.arrow.flatbuf.TimeUnit.NANOSECOND);
+  SECOND(org.apache.arrow.flatbuf.TimeUnit.SECOND, java.util.concurrent.TimeUnit.SECONDS),
+  MILLISECOND(org.apache.arrow.flatbuf.TimeUnit.MILLISECOND, java.util.concurrent.TimeUnit.MILLISECONDS),
+  MICROSECOND(org.apache.arrow.flatbuf.TimeUnit.MICROSECOND, java.util.concurrent.TimeUnit.MICROSECONDS),
+  NANOSECOND(org.apache.arrow.flatbuf.TimeUnit.NANOSECOND, java.util.concurrent.TimeUnit.NANOSECONDS);
```

Comment:
Would it be better to leave this as it was and just put a switch statement in `getTimeUnit` to return the corresponding java TimeUnit?

Reply:
Not sure why that's better...Why do you think so? Maybe I missed sth.

Reply:
Well before the enum was defined by the flatbufId, but here it is the flatbufId and the Java TimeUnit which is redundant information.  So a check for equality would have to look at both of these fields.  Maybe not a big deal though..

Reply:
Enum equality should just be identity check:
https://stackoverflow.com/questions/533922/is-it-ok-to-use-on-enums-in-java